File: Python/Codeforces/1971_E.py
from bisect import bisect_left

for i in range(1, int(input()) + 1):
    n, k, q = map(int, input().split())
    
    signs = [0]
    minutes = [0]
    
    signs.extend(list(map(int, input().split())))
    minutes.extend(list(map(int, input().split())))
    
    while q > 0:
        d = int(input())
        
        pos = bisect_left(signs, d) - 1
        
        mins = int(minutes[pos] + ((d - signs[pos]) * (minutes[pos + 1] - minutes[pos]) // (signs[pos + 1] - signs[pos])))
        
        print(f"{mins}", end=" ")
    
        q -= 1
    
    print()

File: CPP/Other CF Problems/1971_E.cpp
#include <bits/stdc++.h>

using namespace std;

void solve()
{
    int n, k, q, d, i, pos, mins;
    
    cin >> n >> k >> q;

    int signs[k + 1];
    int minutes[k + 1];

    signs[0] = minutes[0] = 0;

    for (i = 1; i <= k; i++) {
        cin >> signs[i];
    }

    for (i = 1; i <= k; i++) {
        cin >> minutes[i];
    }

    while (q--) {
        cin >> d;

        pos = lower_bound(signs + 1, signs + k + 1, d) - signs - 1;

        mins = minutes[pos] + (d - signs[pos]) * 1.0 * (minutes[pos + 1] - minutes[pos]) / (signs[pos + 1] - signs[pos]);

        cout << mins << " ";
    }

    cout << "\n";
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;

    cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}